app/services/owner_classifier.rb

```diff
@@ -1,10 +1,30 @@
 class OwnerClassifier
-  def initialize(id_and_type)
-    @id_and_type = id_and_type
-  end
+  class << self
+    def classify_owner(owner_id_and_type)
+      return nil unless owner_id_and_type
+
+      validate_input_format! unless valid_input?(owner_id_and_type)
+      owner = find_owner_by_id(owner_id_and_type)
+      raise_owner_not_found_error! unless owner
+      owner
+    end
+
+    private
+
+    def raise_owner_not_found_error!
+      raise ActiveRecord::RecordNotFound, _("User or Usergroup not found.")
+    end
+
+    def validate_input_format!
+      raise ArgumentError, _("Invalid input format. Please use the format '${id}-[Users|Usergroups]'.")
+    end
+
+    def find_owner_by_id(id_and_type)
+      id_and_type.include?('Users') ? User.find_by_id(id_and_type.to_i) : Usergroup.find_by_id(id_and_type.to_i)
+    end
 
-  def user_or_usergroup
```

Comment:
You're sure this isn't used anywhere? Normally we add a deprecation and remove it in a future release. Mostly for plugins.

Reply:
I didn't find any other uses in Foreman or Katello. 
I can keep the old method name, but since we are changing it to a static method, we would still need to make adjustments.

Reply:
Why not adding a static method in addition to the instance one:

``` ruby
class OwnerClassifier
  def self.classify_owner(owner_id_and_type)
    return self.new(owner_id_and_type).user_or_usergroup
  end

  def user_or_usergroup
    # ...
  end
end
```

This will keep the old way and also add the static shortcut. It will also be much more testable.

Reply:
@ShimShtein, could you check the recent changes and confirm if they match your suggestion?